Modules/posixmodule.c

```diff
@@ -5309,52 +5311,68 @@ os_spawnve_impl(PyObject *module, int mode, path_t *path, PyObject *argv,
 
 
 #ifdef HAVE_FORK
+
+/* Helper function to validate arguments.
+   Returns 0 on success.  1 on failure with a TypeError raised.
+   If obj is non-NULL it must be callable.  */
+static int
+check_null_or_callable(PyObject *obj, const char* obj_name)
+{
+    if (obj && !PyCallable_Check(obj)) {
+        PyErr_Format(PyExc_TypeError, "'%s' must be callable, not %R",
```

Comment:
The error message looks like:
```
'before' must be callable, not <class 'str'>
```
It is more common to include the type name rather than repr.
```
'before' must be callable, not str
```